src/features/counterfactual/utils.ts

```diff
@@ -220,3 +220,90 @@ export const showSubmitNotification = (dispatch: AppDispatch, chain?: ChainInfo,
     }),
   )
 }
+
+// TODO: Reuse this for safe creation flow instead of checkSafeCreationTx
```

Comment:
There's a lot of duplicate code indeed.

Reply:
Yeah, I tried reusing the existing code but it has very specific types that are too coupled to the safe creation flow. I plan to reuse this new EventBus to refactor the existing safe creation flow eventually.

Reply:
I meant that in this very PR there are two copies of the same routines, down to exact same error messages.

Reply:
The only difference is that instead of returning the value or calling a setter we call `safeCreationDispatch`. I would prefer we copy it for now and refactor it later as I don't want to change the safe creation flow logic in the scope of this PR.